### ISSUE TITLE
Creating new links or editing existing ones is impossible

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5#6530]
## 📝 Provide detailed reproduction steps (if any)

1. Select some text.
2. Click link
3. Type some text.
4. Save link.

### ✔️ Expected result

Link is created.

### ❌ Actual result

The balloon says "This link has no URL". The model attribute value is also empty.

## 📃 Other details

This is a regression on master after https://github.com/ckeditor/ckeditor5-link/pull/256.

**Since tests didn't catch it, they're either too weak or some test is missing.**

cc @panr @jodator 

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by panr** [comment]
I'll check it tomorrow 👍

**Comment by panr** [comment]
I've fixed the changes I made in ckeditor/ckeditor5-link#256.

Related question about preventing this issue in the future: why do we allow to submit empty URL? 🤔

**Comment by oleq** [comment]
> why do we allow to submit empty URL

Mostly because it simplified the design and implementation of the link UI. If an empty URL was forbidden, we'd need to:
* display an error,
* prevent the balloon from disappearing until a valid URL is typed in (???), 
   * this is hard because by design all pop-ups in the UI are dismissible when you click somewhere else
   * or... maybe we don't have to do that and just do nothing to the model if there was no URL and the pop-up was dismissed (???)

Long story short, this allowed us to speed up the development of the first (or one of the first) features a few years ago. Besides, a `href`-less anchor is a valid anchor:
https://stackoverflow.com/questions/10510191/valid-to-use-a-anchor-tag-without-href-attribute

But you're right. Maybe we should revisit this design. If you have some ideas how to do it right, please create a follow-up with the proposal.
